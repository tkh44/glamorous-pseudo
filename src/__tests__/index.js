/* eslint-env jest */
/* eslint-disable jsx-quotes */
import React from 'react'
import renderer from 'react-test-renderer'
import { matcher, serializer } from 'jest-glamor-react'
import glamorous from 'glamorous'
import Pseudo from '../index'

expect.addSnapshotSerializer(serializer)
expect.extend(matcher)

describe('restyles', () => {
  test('renders elements with styles', () => {
    const tree = renderer
      .create(
        <Pseudo
          after={{ content: '" after"', color: 'red' }}
          before={{ content: '"before "', color: 'green' }}
          focus={{ color: 'blue' }}
          hover={{ color: 'purple' }}
          active={{ fontSize: '0.8em' }}
          visited={{ color: '#5EDA7E' }}
          display="block"
          width={300}
          textAlign="center"
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshotWithGlamor()
  })

  test('renders elements with styles', () => {
    const tree = renderer
      .create(
        <Pseudo
          after={{ content: '" after"', color: 'red' }}
          component={glamorous.H1}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshotWithGlamor()
  })

  test('renders elements with styles', () => {
    const tree = renderer
      .create(
        <Pseudo hover={{ '& div': { color: 'blue' } }} component={glamorous.H1}>
          <h1>Blue H1 on hover</h1>
        </Pseudo>
      )
      .toJSON()

    expect(tree).toMatchSnapshotWithGlamor()
  })
})
