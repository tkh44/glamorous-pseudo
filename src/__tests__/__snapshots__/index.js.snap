// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restyles renders elements with styles 1`] = `
.css-4h4i2w:active,
[data-css-4h4i2w]:active {
  font-size: 0.8em;
}

.css-4h4i2w:after,
[data-css-4h4i2w]:after {
  content: " after";
  color: red;
}

.css-4h4i2w:before,
[data-css-4h4i2w]:before {
  content: "before ";
  color: green;
}

.css-4h4i2w:focus,
[data-css-4h4i2w]:focus {
  color: blue;
}

.css-4h4i2w:hover,
[data-css-4h4i2w]:hover {
  color: purple;
}

.css-4h4i2w:visited,
[data-css-4h4i2w]:visited {
  color: #5EDA7E;
}

<div
  className="css-4h4i2w"
/>
`;

exports[`restyles renders elements with styles 2`] = `
.css-5c4x45:after,
[data-css-5c4x45]:after {
  content: " after";
  color: red;
}

<h1
  className="css-5c4x45"
/>
`;

exports[`restyles renders elements with styles 3`] = `
.css-42g7tz:hover div,
[data-css-42g7tz]:hover div {
  color: blue;
}

<h1
  className="css-42g7tz"
>
  <h1>
    Blue H1 on hover
  </h1>
</h1>
`;
